src/borg/helpers/progress.py

```diff
@@ -24,9 +25,15 @@ def __init__(self, msgid=None):
         self.id = self.operation_id()
         self.msgid = msgid
 
-    def make_json(self, *, finished=False, **kwargs):
+    def make_json(self, *, finished=False, override_time: typing.Optional[float] = None, **kwargs):
```

Comment:
Let me know if you think this new argument is bad. The function is harder to test when it always takes `time.time()`.